contrib/platform/test/com/sun/jna/platform/win32/WinPerfTest.java

```diff
@@ -0,0 +1,148 @@
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Test;
+
+import com.sun.jna.Memory;
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+import com.sun.jna.platform.win32.WinDef.DWORDByReference;
+import com.sun.jna.platform.win32.WinPerf.PERF_COUNTER_BLOCK;
+import com.sun.jna.platform.win32.WinPerf.PERF_COUNTER_DEFINITION;
+import com.sun.jna.platform.win32.WinPerf.PERF_DATA_BLOCK;
+import com.sun.jna.platform.win32.WinPerf.PERF_INSTANCE_DEFINITION;
+import com.sun.jna.platform.win32.WinPerf.PERF_OBJECT_TYPE;
+import com.sun.jna.ptr.IntByReference;
+
+/**
+ * Tests structures in WinPerf
+ * 
+ * @author dbwiddis
+ */
+public class WinPerfTest extends AbstractWin32TestSupport {
+    /*-
+     * This method tests: PERF_DATA_BLOCK, PERF_OBJECT_TYPE, PERF_COUNTER_DEFINITION,
+     * PERF_INSTANCE_DEFINITION, and PERF_COUNTER_BLOCK
+     *
+     * See Performance Data format at
+     * https://msdn.microsoft.com/en-us/library/windows/desktop/aa373105(v=vs.85).aspx
+     * [ ] Perf_Data_block
+     * Then one or more objects repeating everything below:
+     * [ ] Object Type
+     * [ ][ ][ ] Multiple counter definitions
+     * Then multiple instances repeating:
+     * [ ] Instance Definition
+     * [ ] Instance name
+     * [ ] Counter Block
+     * [ ][ ][ ] Counter data for each definition above
+     */
+    @Test
+    public void testPerfDataBLock() {
+        if (AbstractWin32TestSupport.isEnglishLocale) {
+            // To test the structures we'll get the data block for Process
+            // performance data
+            DWORDByReference index = new DWORDByReference();
+            Pdh.INSTANCE.PdhLookupPerfIndexByName(null, "Process", index);
+            int processIndex = index.getValue().intValue();
+            String processIndexStr = Integer.toString(processIndex);
+            // And one of the counters, for PID:
+            Pdh.INSTANCE.PdhLookupPerfIndexByName(null, "ID Process", index);
+            int idProcessIndex = index.getValue().intValue();
+
+            // now load the Process registry entry
+            // Sequentially increase the buffer until everything fits.
+            int perfDataBufferSize = 4096;
+            IntByReference lpcbData = new IntByReference(perfDataBufferSize);
+            Pointer pPerfData = new Memory(perfDataBufferSize);
+            while (WinError.ERROR_MORE_DATA == Advapi32.INSTANCE.RegQueryValueEx(WinReg.HKEY_PERFORMANCE_DATA,
+                    processIndexStr, 0, null, pPerfData, lpcbData)) {
+                perfDataBufferSize += 4096;
+                lpcbData.setValue(perfDataBufferSize);
+                pPerfData = new Memory(perfDataBufferSize);
+            }
+            // When this loop exits, pPerfData points to the PERF_DATA_BLOCK.
+
+            // PERF_DATA_BLOCK test
+            // Test the signature, a char[] containing [P, E, R, F]
+            PERF_DATA_BLOCK perfData = new PERF_DATA_BLOCK(pPerfData.share(0));

```

Comment:
Why the `share(0)` and not directly pass in the `pPerfData`? Is this for consistency with the following `share` calls?

Reply:
My intent was to highlight the fact that the `PERF_DATA_BLOCK` is only the first portion of the allocated `pPerfData` memory.  So there was somewhat of a consistent approach with that vs. the other calls.   I agree that it's not necessary, however, and I'll remove it.